katz/src/main/kotlin/katz/instances/Composed.kt

```diff
@@ -0,0 +1,54 @@
+package katz
+
+/**
+ * https://www.youtube.com/watch?v=wvSP5qYiz4Y
+ */
+interface ComposedType<out F, out G>
```

Comment:
It cannot be private, the compiler marks other instances that expose it like the constructor of the composed types. It also disallows lower(), which is used by the classes that use the composed constructs: https://github.com/FineCinnamon/Katz/blob/c4f790b9dff9201a82d338b9520860427e3289cf/katz/src/main/kotlin/katz/data/EitherT.kt#L72